content/knowledge-others/install-unity-version.md

```diff
@@ -99,32 +99,93 @@ Use the Unity version you installed on the machine:
+
+1. Open your Codemagic app settings, and go to the **Environment variables** tab.
+2. Enter the desired **_Variable name_**, e.g. `AWS_ACCESS_KEY_ID`.
+3. Enter the required value as **_Variable value_**.
+4. Enter the variable group name, e.g. **_aws_credentials_**. Click the button to create the group.
+5. Make sure the **Secure** option is selected.
+6. Click the **Add** button to add the variable.
+7. Repeat the process to also add the `AWS_SECRET_ACCESS_KEY` variable.
+
+8. Import the  **_aws_credentials_** group and install the **awscli** using this script.
 
-{{< tabpane >}}
-{{% tab header="Mac" %}}
 {{< highlight yaml "style=paraiso-dark">}}
-    cache:
-      cache_paths:
-        - /Applications/Unity/Hub/Editor/${UNITY_VERSION}
+environment:
+  groups:
+    - aws_credentials
+scripts:
+  - name: Install awscli
+    script: | 
+      sudo pip3 install awscli --upgrade
 {{< /highlight >}}
-{{< /tab >}}
-{{% tab header="Windows" %}}
+
+Add the script below to your `scripts` section before your build script to check if S3 bucket has an old cached file. 
+
 {{< highlight yaml "style=paraiso-dark">}}
-    cache:
-      cache_paths:
-        - C:\Program Files\Unity\Hub\Editor\$UNITY_VERSION
+- name: Check S3 bucket
+  script: | 
+    if echo $(aws s3 ls s3://<BUCKET_NAME>) | grep -q ${UNITY_VERSION}.tar.gz ; then 
+      echo "Unity editor ${UNITY_VERSION} was found in the S3 bucket."; 
+      echo "S3_HAS_EDITOR=true" >> $CM_ENV
+    else 
+      echo "Unity editor ${UNITY_VERSION} was not found in the S3 bucket."; 
+      echo "S3_HAS_EDITOR=false" >> $CM_ENV
+    fi
 {{< /highlight >}}
-{{< /tab >}}
 
-{{< /tabpane >}}
+{{<notebox>}}
+Replace `<BUCKET_NAME>` with your actual bucket name.
+{{</notebox>}}
+
+Now you can check the `$S3_HAS_EDITOR` variable, if the S3 bucket has the editor then just copy it to the build machine then extract it, or install the Unity editor with the modules then compress it and upload it to S3.
+
+{{< highlight yaml "style=paraiso-dark">}}
+- name: Install Unity editor
+  script: | 
+    if [ "$S3_HAS_EDITOR" == "false" ] ; then
+      /Applications/Unity\ Hub.app/Contents/MacOS/Unity\ Hub -- --headless install --version ${UNITY_VERSION} --changeset ${UNITY_VERSION_CHANGESET}
+      /Applications/Unity\ Hub.app/Contents/MacOS/Unity\ Hub -- --headless install-modules --version ${UNITY_VERSION} -m ios android android-sdk-ndk-tools android-open-jdk
+    fi
+  ignore_failure: true
+- name: Compress Unity editor
+  script: | 
+    if [ "$S3_HAS_EDITOR" == "false" ] ; then
+      cd /Applications/Unity/Hub/Editor/
+      tar -cv ${UNITY_VERSION}/ | gzip > ${UNITY_VERSION}.tar.gz
+    fi
+- name: Copy editor to S3 bucket
+  script: | 
+    if [ "$S3_HAS_EDITOR" == "false" ] ; then
+      aws s3 cp /Applications/Unity/Hub/Editor/${UNITY_VERSION}.tar.gz s3://<BUCKET_NAME>
+    fi
```

Comment:
Should this be done at the end of the workflow, in the Publishing section scripts, if required?